Assets/_BForBoss/_Core/Scripts/TimerViewBehaviour.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using TMPro;
+using UnityEngine;
+
+namespace BForBoss
+{
+    public class TimerViewBehaviour : MonoBehaviour
+    {
+        private TMP_Text _timerLabel;
+        private float _time;
+        private bool _isRunning;
+
+        public void StartTimer()
+        {
+            _isRunning = true;
+        }
+
+        public void StopTimer()
+        {
+            _isRunning = false;
+        }
+
+        public void Reset()
+        {
+            _time = 0;
+            _isRunning = true;
+            _timerLabel.text = string.Empty;
+        }
+
+        private void Update()
+        {
+            if (_isRunning)
+            {
+                _time += Time.deltaTime;
+                SetTimerLabel(_time);
+            }
+        }
+
+        private void SetTimerLabel(float seconds)
+        {
+            TimeSpan time = TimeSpan.FromSeconds(seconds);
+            string str = time .ToString(@"mm\:ss\:fff");
+            _timerLabel.text = str;
+        }
+ 
+        private void Awake()
+        {
+            #if !(UNITY_EDITOR || DEVELOPMENT_BUILD)
+                Destroy(gameObject);
+            #endif
+            _timerLabel = GetComponentInChildren<TMP_Text>();
```

Comment:
Wouldn't this cause a problem (i.e. a null reference exception or something) in a production build? since we are trying to set `_timerLabel ` after destroying the object?? Or am I crazy and this is handled just fine

Reply:
If it's Destroyed, the rest wouldn't be ran

Reply:
https://answers.unity.com/questions/637668/when-i-destroy-my-gameobject-my-script-stops-on-wo.html